delete-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusDeleteHandler.java

```diff
@@ -0,0 +1,125 @@
+package no.sikt.nva.scopus;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.S3Event;
+
+import java.net.URI;
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Lists;
+import no.unit.nva.s3.S3Driver;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.paths.UriWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.eventbridge.EventBridgeClient;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequest;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import static nva.commons.core.attempt.Try.attempt;
+
+public class ScopusDeleteHandler implements RequestHandler<S3Event, Void> {
+
+    public static final int SINGLE_EXPECTED_RECORD = 0;
+    public static final int PUT_EVENTS_REQUEST_MAX_ENTRIES = 2;
+    public static final String S3_SCHEME = "s3";
+    public static final String DELETE_SCOPUS_IDENTIFIER_PREFIX = "DELETE-2-s2.0-";
+    public static final String EVENT_TOPIC_IS_SET_IN_EVENT_BODY = "ReferToEventTopic";
+    public static final String EMPTY_STRING = "";
+    private static final Logger logger = LoggerFactory.getLogger(ScopusDeleteHandler.class);
+    private static final String EVENT_BUS = new Environment().readEnv("EVENTS_BUS");
+    private final S3Client s3Client;
+    private final EventBridgeClient eventBridgeClient;
+
+    @JacocoGenerated
+    public ScopusDeleteHandler() {
+        this(S3Driver.defaultS3Client().build(), EventBridgeClient.create());
+    }
+
+    public ScopusDeleteHandler(S3Client s3Client, EventBridgeClient eventBridgeClient) {
+        this.s3Client = s3Client;
+        this.eventBridgeClient = eventBridgeClient;
+    }
+
+    @Override
+    public Void handleRequest(S3Event event, Context context) {
+        List<String> identifiersToDelete = readIdentifiersToDelete(event);
```

Comment:
Collecting the identifiers and creating a new list of identifiers might lead to memory overflow. But we can deal with this later.

Reply:
Ideally, we should group the items in the stream and send an event without needing to read the whole list to begin with